### DISCUSSION TITLE
Colours aren't converted?

### DISCUSSION THREAD

**Discussion by yea233** [BadDogSkyrim/PyNifly#222] [Announcements]
(warning: NSFW image)
i don't know if this is a bug at all but I don't know where to post but here :c since im very new to this I feel like it's probably me not knowing whats the problem than it being actually a bug. despite few errors I can export a NIF but the colours or textures never get forwarded with the NIF?
I'm trying to convert him:
![IMG_20230218_192001](https://user-images.githubusercontent.com/125409111/219879665-7e8afbd2-1618-4d95-a28c-f85725a6d100.png)
I have the NIF and all, but all the colours are gone, his download page said that the textures/colours are already in the blend itself, and on blender there's no shader tab, can someone please help? I also found some TRI files or MTL, are these relevant to colours?

**Comment by BadDogSkyrim** [comment]
So now you get to learn all about shaders and textures and such.

First you have to to figure out where the texturing is coming from. It could be that there are embedded texture files or it could be that everything is generated through shader nodes. The greek-key type patterns are probably texture files, but the glow on the hammer might come from shader nodes. 

Since you don't have a shader tab, just switch to the shader editor in any convenient pane and see what's there.

If it's all nodes, you'll probably want to bake textures from the nodes to image files. If it's all files, you just need to save them outside blender and convert to the right format. 

If everything is set up right, the exporter will export everything you need in the nif. The easiest way to get everything set up right is to import a similar nif, assign the material to your shape, and tweak the material.

The glow on the hammer can be reproduced, but how you do it depends on the game you're exporting for.

**Comment by yea233** [comment]
> So now you get to learn all about shaders and textures and such.
> 
> First you have to to figure out where the texturing is coming from. It could be that there are embedded texture files or it could be that everything is generated through shader nodes. The greek-key type patterns are probably texture files, but the glow on the hammer might come from shader nodes.
> 
> Since you don't have a shader tab, just switch to the shader editor in any convenient pane and see what's there.
> 
> If it's all nodes, you'll probably want to bake textures from the nodes to image files. If it's all files, you just need to save them outside blender and convert to the right format.
> 
> If everything is set up right, the exporter will export everything you need in the nif. The easiest way to get everything set up right is to import a similar nif, assign the material to your shape, and tweak the material.
> 
> The glow on the hammer can be reproduced, but how you do it depends on the game you're exporting for.

hey thank you for replying.. I think it's a bit complicated, since I'm not used to Blender yet, but I'll try to follow the instructions. I'm converting all this to Skyrim btw, don't know if it will work. the nifs are there for many models all that is left is the textures since they never exported(and also bones according to Skyrim? don't know I'll figure it out)

let's say something easier, if the textures already exist as files in a folder with the blender file and the converted NIF, they're all in PNG. I have to convert them alone right? this wouldn't just export them with the NIF?
 I have Substance Pai but I can't find a clear up to date guide on how to convert them to DDS. Edit: nvm I found a guide and deleted substance paint since it's too intense and I just didn't know how to use it.

btw sorry for the questions, you don't have to answer any, it was just some things I was wondering, your tool works great, it converted many blender files into NIFS (albeit without colours probably cause idk what to do lol), thank you

**Comment by BadDogSkyrim** [comment]
Yeah, and once you're done with colors if you want the guy to do anything you'll have to rig him to a skeleton and that's another can of worms. At least if you want him to be anything more than a statue in game.

I encourage you to go ahead--it's all doable--but expect to have to come up to speed on a bunch of different tools and processes.

One thing to do is import a body or armor nif from your target game and explore. See how it's set up in blender after import. That's how you'll want your guy to be set up for export.

**Comment by podvornyakva** [comment]
Modified images can be saved by blender. Put *.png somewhere. Load in blender. Paint over. In "image" window press image->save image. That is it. Blender DO NOT support *.dds format, cause of PROPRIETARY license. Blender do not support ANY PROPRIETARY shit. So you need to convert texture images by hands, with plugin for Photoshop or Gimp. After that you can modify *.BGSM material file, if it necessary(check nexus for material editor). That's it - that how it works.
If you have unpacked all Skyrim data files, in corresponding folder structure, textures automatically loads on import.
Also about dds. There is a bunch of encoding algorithms:
BS1/DXT1 - for masks, gradients and palettes.
BS3/DXT5 - mostly for base color(diffuse/albedo) images.
BS5/ATI2 - for normal maps and other combinations of 3 channel images (specular(R)+roughness(G)+metallness(B) and so on.)
Do not recommend to use BS7. Lot of software incompatible with that encoding algorithm.

It is very complicate task to add completely new character.  First you have to import nif with similar creature. Put your mesh on imported skeleton. Transfer vertex/weight groups, shapemorfs from imported mesh to minoth. Bake images. Export back meshes, in corresponding folder structure and same name with original nifs. Put own images same way. Launch the game and find creature you replaced. Creation of very own mod is even more complicate task. Learn, explore, ask and you will understand what is need to be done to achieve goal. Start with simple tasks. Then go complicate. Try to modify textures of existing one creature/object, to understand workflow of texture import/export. Once it become clear for you - go further, and try to create simple mod that change texture without replacing original. And so on...
